Assets/Scripts/Models/Area/RoomManager.cs

```diff
@@ -0,0 +1,427 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ProjectPorcupine.Rooms
+{
+    public class RoomManager : IEnumerable<Room>
+    {
+        /// <summary>
+        /// A all the currently managed rooms.
+        /// </summary>
+        private List<Room> rooms;
+
+        public RoomManager()
+        {
+            rooms = new List<Room>();
+
+            // Create the outside room.
+            OutsideRoom = new Room();
+
+            // Add the outside rooom to the list just in case.
+            rooms.Add(OutsideRoom);
+        }
+
+        #region Events
```

Comment:
Maybe we should look to see if more regions are better... we could maybe do an Issue to find what the "sweet" spot for this might be